resources/aws_ec2.go

```diff
@@ -677,3 +681,27 @@ func (obj *AwsEc2Res) UnmarshalYAML(unmarshal func(interface{}) error) error {
 func (obj *AwsEc2Res) prependName() string {
 	return AwsPrefix + obj.GetName()
 }
+
+// snsServer returns an http server used to listen for sns messages.
+func (obj *AwsEc2Res) snsServer() *http.Server {
+	addr := obj.WatchListenAddr
+	// if addr is a port
+	if _, err := strconv.Atoi(obj.WatchListenAddr); err == nil {
+		addr = fmt.Sprintf(":%s", obj.WatchListenAddr)
+	}
+	srv := &http.Server{Addr: addr}
+	log.Printf("%s: Starting SNS Endpoint", obj)
+	http.HandleFunc("/", obj.snsPostHandler)
```

Comment:
weird that this HandleFunc isn't associated with the particular server at all... Is this right? How does this work? What if we had two different http servers running on different ports? isn't that allowed?

Reply:
https://golang.org/src/net/http/server.go?#L2345 is needed AFAICT